drivers/flash/nrf_qspi_nor.c

```diff
@@ -479,13 +479,13 @@ static inline int qspi_nor_read_id(struct device *dev,
 static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 			 size_t size)
 {
-	if (!dest) {
+	/* destination must be aligned to a 32-bit word. */
+	if ((dest == NULL) || (((uintptr_t)dest % sizeof(uint32_t)) != 0)) {
```

Comment:
nit: u32_t ? I see this is mostly used in this file as well.

Reply:
Also, what is the gain of casting the `dest` to `uintptr_t` ?

Reply:
Isn't `void *` wider pointre type than `uintptr_t`?

Reply:
Would `nrfx_is_word_aligned` work as a check here?

Reply:
Switched to `u32_t` (original incorrectly had `uint32_t`, didn't notice).

`dest` is a pointer.  Modulo operation on a pointer is not defined; the pointer must be cast to an integer.  The only integral types to which pointers can legally be cast are `intptr_t` and `uintptr_t`, and we don't need signed modulo so `uintptr_t` is correct.

I was unaware of `nrfx_is_word_aligned` and don't see it used elsewhere in Zephyr.